### DISCUSSION TITLE
Why to disable flow control?

### DISCUSSION THREAD

**Discussion by kvaps** [ltsp/ltsp#214] [Q&A]
Hi, I just found that LTSP [disables flow control](https://github.com/ltsp/ltsp/blob/master/ltsp/common/service/55-service.sh#L33) for all physical interfaces

Could you explain is there specific reason for that?

> We want to disable flow control in all interfaces, as all of them might serve LTSP clients.

This comment is not much informative and does not describes the real need of this procedure. I would like to know why is this needed for.

Thank you

**Comment by alkisg** [comment]
The theory is explained in Wikipedia's [Flow control > issues > head-of-line blocking](https://en.wikipedia.org/wiki/Ethernet_flow_control#Issues) paragraph.

If the server <=> switch connection is gigabit, and even one client or printer is 100 Mbps, then that client may send a pause signal to the server, which will prevent the server from sending to **all** clients for a while. Essentially, this makes gigabit networks behave like 100 Mbps networks.

If you do a network benchmark with Epoptes with flow control enabled, you see the server sending 100 Mbps total, while with flow control disabled, 1000 Mbps total. That's 10 times faster.

Managed switches usually default to disabled flow control. Unmanaged switches default to enabled, and support disabling it during link negotiation, which is what our code does. So our code solves most of the problems, and that's why we don't need to document flow control much.

What we can't help with is unmanaged chained switches:

- Server connects to switch1 with gigabit link
- Switch1 connects to switch2 with gigabit link
- In switch2 there are nine gigabit clients and one 100 Mbps client

In that case, the 100 Mbps client will send a pause frame to switch2, which we don't have control over, and it will degrade our gigabit network to 100 Mbps network.

This is why people buying new switches should prefer managed switches that default to disabled flow control, and use them as the "central" switch1 in the example above. One cheap and good such switch is the [TP-Link TL-SG1024DE ](https://www.tp-link.com/us/business-networking/easy-smart-switch/tl-sg1024de/).

**Comment by kvaps** [comment]
Thanks for detailed explanation, do I right understand that flow control should be disabled on both: server and clients?

**Comment by alkisg** [comment]
Yes, it would help the LTSP clients in the case that a client would need to send a lot of data to other clients. For example, if a client had a gigabit network connection and needed to broadcast its screen to other clients, and some of those other clients had 100 Mbps connection.
